### PR TITLE
added graph traversals

### LINKED ISSUE #24: Implement graph alogrithms
- [ ] 1. Depth First Traversal (DFS)
- [ ] 2. Breadth First Traversal (BFS)

### PR DESCRIPTION
**Pull request by rootaux**
### Description

I have implemented BFS and DFS graph traversals method in Python 3. Both methods return a list which gives the path followed by the algorithm. 

### Related Issue

** Please don't create a Pull request without creating an issue first **

Fixes #24 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)

### How Has This Been Tested?

I have tested the programs with some manual inputs. I have yet to write the test file. 

### Checklist:

- [x] I read the **CONTRIBUTING** document before making this pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] My change requires change to documentation
- [ ] I have updated documentation accordingly
- [ ] I have updated the requirements file
- [x] My changes generate no new warnings 
- [x] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by devkapilbansal** [comment]
@codesankalp please review it once

**Comment by devkapilbansal** [comment]
@rootaux it would be better if you add required test cases too

File: dsalgo/graph_traversal.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		##Add the current node to the visited array
+		visited.append(s)
+		if ans is None:
+			ans = []
+		ans.append(s)
+		for i in self.graph[s]:
+			if i not in visited:
+				self.dfs(i, visited, ans)
+		##Return the ans list
+		return ans
```
Comment (devkapilbansal) [inline dsalgo/graph_traversal.py:44]: Add a new line here

**Comment by rootaux** [comment]
@devkapilbansal updated as you requested and I have also added a test.
The test is written for the following undirected and directed graphs.
undirected graph
directed graph

File: tests/graph_tests.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from dsalgo.graph_traversal import Graph
+
+mat = [
+    [1, 3],
+    [1, 5],
```
Comment (codesankalp) [inline tests/graph_tests.py:5]: can you use unit tests here according to or guidelines?

File: dsalgo/graph_traversal.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			for i in self.graph[s]:
+				if i not in visited:
+					queue.append(i)
+					visited.append(i)
+		##Return the ans list
+		return ans
+
+	##Perform dfs using recursion
+	def dfs(self, s, visited=None, ans=None):
+		#If visited list is empty create one
```
Comment (codesankalp) [inline dsalgo/graph_traversal.py:32]: add docstrings for the :param:

File: dsalgo/graph_traversal.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+class Graph:
+	def __init__(self):
+		self.graph = defaultdict(list)
+
+	#Method to add an edge to the graph
+	def addEdge(self, u ,v):
+		self.graph[u].append(v)
+
+	##Perform bfs on the graph using queue
+	def bfs(self, s):
```
Comment (codesankalp) [inline dsalgo/graph_traversal.py:11]: add docstrings

File: tests/graph_tests.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    [4, 8],
+    [10, 2]]
+
+g = Graph()
+b = Graph()
+#Test for directed graph
+for i in mat:
+    g.addEdge(i[0], i[1])
+    b.addEdge(i[0], i[1])
+    b.addEdge(i[1], i[0])
```
Comment (codesankalp) [inline tests/graph_tests.py:25]: reason for this?

Comment (rootaux) [reply tests/graph_tests.py:25]: G is directed graph and B is undirected graph that means if you can go from A to B then you can also go from B to A

File: tests/graph_tests.py (hunk 2 above)
Comment (rootaux) [reply tests/graph_tests.py:5]: I'm not sure if unit test would work here, because BFS or DFS could have multiple correct answers.

Comment (devkapilbansal) [reply tests/graph_tests.py:5]: Add two or three tests of both dfs and bfs, you can take any or more of those multiple correct answers to work on.

**Comment by Aryamanz29** [comment]
Closing this PR, Since contributor is inactive
Thanks

**Pull request closed without merging**